overlord/ifacestate/handlers.go

```diff
@@ -1025,6 +1035,16 @@ func (m *InterfaceManager) doAutoConnect(task *state.Task, _ *tomb.Tomb) error {
 		return fmt.Errorf("auto-connect conflict check failed: %v", err)
 	}
 
+	// Consider gadget connections
+	if err := gadgectConnect.addGadgetConnections(newconns, conns, conflictError); err != nil {
+		return err
+	}
+	connOpts := make(map[string]*connectOpts, len(newconns))
```

Comment:
A little caveat for anyone potentially touching this code later, that perhaps deserves a short comment: this should never be reordered with regard to the code below (when we add more connections to newconns), else it would apply ByGadget to all of them. Perhaps connOpts should be populated by addGadgetConnections() as it updates newconns?

Reply:
len(newconns) is probably generally small at this point, so that change should be ok if you think it makes things a bit more robust. But is also true that we want to have things with ByGadget set in any case so I should probably either way add a comment that gadget connections should be considered before other auto-connections